generators/client/templates/react/src/main/webapp/app/shared/util/data-utils.ts.ejs

```diff
@@ -0,0 +1,66 @@
+  /**
+   * Method to open file
+   */
+  export const openFile = (contentType: string, data: string) => () => {
+    const fileURL = `data:${contentType};base64,${data}`;
+    const win = window.open();
+    win.document.write(
+      '<iframe src="' + fileURL + '" frameborder="0" style="border:0; top:0px; left:0px; bottom:0px; right:0px; width:100%; height:100%;" allowfullscreen></iframe>'
+    );
+  };
+
+  /**
+   * Method to convert the file to base64
+   */
+  const toBase64 = (file: File, cb: Function) => {
+    const fileReader: FileReader = new FileReader();
+    fileReader.readAsDataURL(file);
+    fileReader.onload = e => {
+      const base64Data = e.target['result'].substr(e.target['result'].indexOf('base64,') + 'base64,'.length);
+      cb(base64Data);
+    };
+  };
+
+  /**
+   * Method to clear the input
+   */
+  export const clearInputImage = (component, elementRef: string) => {
+    if (component && elementRef) {
+      if (component.state[elementRef]) {
+        component.setState({
+          [elementRef]: null
+        });
+      }
+    }
+  };
+
+  export const endsWith = (suffix: string, str: string): boolean => str.indexOf(suffix, str.length - suffix.length) !== -1;
```

Comment:
why dont you use the string endsWith method? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith

Reply:
The whole file is a copy paste from the [angular lib](https://github.com/jhipster/ng-jhipster/blob/master/src/service/data-util.service.ts). So I haven't really paid attention to what was inside. It might be a good idea to apply that fix to the angular lib too.